.circleci/config.yml

```diff
@@ -0,0 +1,70 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+version: 2.1
+
+commands:
+  build-common:
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - maven-repo-v1-{{ .Branch }}-{{ checksum "pom.xml" }}
+            - maven-repo-v1-{{ .Branch }}-
+            - maven-repo-v1-
+      - save_cache:
```

Comment:
I think we should save the cache after the build, right? We should also exclude zeppelin artifacts (See [GitHub-Action](https://github.com/apache/zeppelin/blob/79750a88025bd64cff5830cf2ec8e6a6181454f5/.github/workflows/core.yml#L51)).

Reply:
Right!
It seems CircleCI does not support `!...` syntax, so I added a new step that deletes `~/.m2/repository/org/apache/zeppelin`.
Since we use `mvn verify` instead of `mvn install` this folder should never be there, but we delete it just in case!